pkgs/test/cc-wrapper/default.nix

```diff
@@ -46,6 +46,17 @@ in stdenv.mkDerivation {
       $READELF -d ./atomics.so | grep libatomic.so && echo "ok" >&2 || echo "failed" >&2
     ''}
 
+    # Test that linking libc++ works, and statically.
+    ${lib.optionalString isCxx ''
+      echo "checking whether can link with libc++... " >&2
+      $CXX ${./cxx-main.cc} -c -o cxx-main.o
+      $CC cxx-main.o -lc++ -o cxx-main
+      $CC cxx-main.o -l:libc++.a -o cxx-main-static
+      ${emulator} ./cxx-main
+      ${emulator} ./cxx-main-static
+      rm cxx-main{,-static,.o}
+    ''}
+
```

Comment:
this test fails `nix-build -A tests.cc-wrapper` because it tries to run on vanilla llvm[12-18] and this change only modifies a subset of the llvm16+ packages. eg:

@ofborg build tests.cc-wrapper

(note aarch64-linux fails for other reasons, #285016)

Reply:
`nix-build -A tests.cc-wrapper` also fails on darwin due to the linker not supporting `-l:libc++.a` format flags.

>  checking whether can link with libc++...
> ld: library not found for -l:libc++.a
 > clang-16: error: linker command failed with exit code 1 (use -v to see invocation)

but just verified using the full library path works on x64 linux & x64 darwin.
```diff
- $CC cxx-main.o -l:libc++.a -o cxx-main-static
+ $CC cxx-main.o ${lib.getLib stdenv.cc.libcxx}/lib/libc++.a -o cxx-main-static
```

(PR https://github.com/NixOS/nixpkgs/pull/305876 needed to rebuild darwin stdenv so test timed out building before able to run).

Reply:
I think I've accounted for these two comments now, thank you (please shout if I seem to have missed anything).